### ISSUE TITLE
simpler way to sign my own documents

### ISSUE THREAD

**Issue by Fuseteam** [LibreSign/libresign#4005]
**Is your feature request related to a problem? Please describe.**
There are times when i need to fill out and sign a document

**Describe the solution you'd like**
I would like to be able to sign the document myself

**Describe alternatives you've considered**
request a signature from myself, then open the mail/notification in order to sign the document

**Comment by vitormattos** [comment]
You can do this. Request do yourself to sign the document

**Comment by Fuseteam** [comment]
yes but that workflow seem to be buggy, i seem need to go to notification and the my user seems to be missing at times when i search for myself. and then i ran into an issue where my files disappeared when i tap the "view file" button—

I believe the workflow for signing my own documents could be simpler:
- upload file
- place signature
- sign

rather then
- upload file
- request signature
- search for myself
- press my own name
- place signature
- press request sign
- go to notifaction
- press view
- sign

**Comment by vitormattos** [comment]
I will need by a more clear scenario with details about your environment, administration settings, the detailed steps that you made and what's the unexpected behavior.

**Comment by Fuseteam** [comment]
the scenario is as described, the setup is nextcloud AIO. 

I just assumed that issue happens because it's kinda a roundabout way to sign my own documents, imo it would be a better ux to be more direct; 3 steps compared to 9 as detailed above

**Comment by vitormattos** [comment]
I think that your point is more about UX and not about a bug.

If I get a right understanding, you wish to have a more simple way to sign a document, is this?

**Comment by Fuseteam** [comment]
yes, i would like a simpler way to sign my own documents

**Comment by vitormattos** [comment]
Have many ways to sign a document. But would be good to think more about the UX to make more simple to do a self signature.

**Comment by vitormattos** [comment]
Looking your purpose and talking more about....

We will need to choose what will be the flow or what step of a flow that we will simplify.


Upload file. This could be made at UI by 4 different points or different ways now, I think, from app files have two ways and from LibreSign app have other two ways. 

Place signature. Is necessary to identify who will sign the document, this step can't be removed because can't be you the signer.

Sign. Is possible to sign the document from app files, from notifications, from link received by email, from LibreSign app. Have different start points to sign.

What do you think that could be simplified and what's the suggestion to simplify without removing existing features?

**Comment by vitormattos** [comment]
How to identify if only will be who is requesting a signature that will sign the document?

**Comment by Fuseteam** [comment]
that's what i mean with "sign my own documents" that the person requesting a signature is the one signing the document...................

so i upload a document, i get asked if i want to add a signer or if i want to...............sign it

**Comment by vitormattos** [comment]
This purpose will remove features. Will be necessary think by another approach.

A possible solution in your path is add ad administration settings a flag to disallow to request to sign to others, but this could have a big impact.

**Comment by Fuseteam** [comment]
i don't understand, what features would be removed?

i am not proposing to remove features, i'm proposing to add an additional simpler path that allows me to sign my own documents

**Comment by vitormattos** [comment]
The point is: how to sign by yourself a document without pass by flow to request others to sign ad don't remove the feature to request others to sign the document.

**Comment by vitormattos** [comment]
Is your suggestion to add a new button to sign by yourself like this?


![Screenshot_20241202_125717](https://github.com/user-attachments/assets/08c47185-e4ae-4793-9a30-2e7414dcb80f)

> Since i am the one who is going to sign it, we could introduce a new start point for signing with a new button called "sign".
> 
> the user then gets to place their own signature and a button "sign"

**Comment by Fuseteam** [comment]
yes that is my suggestion

**Comment by vitormattos** [comment]
The button "I will sign" (or other text like this) could be a shortcut to add herself to sign the document.

After click at this button, is the step to choose the place to add the visible signature at the document that is necessary to scroll to the page that you wish to sign, click at the point and after this finish the sign request flow.

Thinking by herself also will be necessary to change the flow after clicking at "I will sign" and this will have more impact that will be necessary to think how to change.

**Comment by vitormattos** [comment]
Other possible suggestion is to add a new option at app Files when we click with left button to automatic add herself as a signer to the document and when click at this option, go to screen to put the signature to the document.

![Screenshot_20241205_121214](https://github.com/user-attachments/assets/f282a78b-2a7f-4290-bf1f-2b14d9650a2d)

**Comment by Fuseteam** [comment]
both approaches sound good to me, tho in my view both have the same flow :thinking:

**Comment by wildernessfamily** [comment]
Having the option `I will sign` #issuecomment-2520592411 would be awesome.
Thank you for all your time and hard work on creating such an amazing and functional plugin for NextCloud 👏

**Comment by vitormattos** [comment]
Maybe `Sign yourself` because I will sign also could be I will wign together with other signers.
